packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -149,124 +194,132 @@ export class TokenRatesController extends StaticIntervalPollingControllerV1<
+   * @param options.currentChainId - The chain ID of the current network.
+   * @param options.currentTicker - The ticker for the current network.
+   * @param options.currentAddress - The current selected address.
+   * @param options.disabled - Boolean to track if network requests are blocked
+   * @param options.tokenPricesService - An object in charge of retrieving token price
+   * @param options.messenger - The controller messaging system
+   * @param options.state - Initial state to set on this controller
    */
-  constructor(
-    {
-      interval = 3 * 60 * 1000,
-      threshold = 6 * 60 * 60 * 1000,
-      getNetworkClientById,
-      chainId: initialChainId,
-      ticker: initialTicker,
-      selectedAddress: initialSelectedAddress,
-      onPreferencesStateChange,
-      onTokensStateChange,
-      onNetworkStateChange,
-      tokenPricesService,
-    }: {
-      interval?: number;
-      threshold?: number;
-      getNetworkClientById: NetworkController['getNetworkClientById'];
-      chainId: Hex;
-      ticker: string;
-      selectedAddress: string;
-      onPreferencesStateChange: (
-        listener: (preferencesState: PreferencesState) => void,
-      ) => void;
-      onTokensStateChange: (
-        listener: (tokensState: TokensState) => void,
-      ) => void;
-      onNetworkStateChange: (
-        listener: (networkState: NetworkState) => void,
-      ) => void;
-      tokenPricesService: AbstractTokenPricesService;
-    },
-    config?: Partial<TokenRatesConfig>,
-    state?: Partial<TokenRatesState>,
-  ) {
-    super(config, state);
-    this.defaultConfig = {
-      interval,
-      threshold,
-      disabled: false,
-      nativeCurrency: initialTicker,
-      chainId: initialChainId,
-      selectedAddress: initialSelectedAddress,
-      allTokens: {}, // TODO: initialize these correctly, maybe as part of BaseControllerV2 migration
-      allDetectedTokens: {},
-    };
+  constructor({
+    interval = DEFAULT_INTERVAL,
+    currentChainId,
+    currentTicker,
+    currentAddress,
+    disabled = false,
+    tokenPricesService,
+    messenger,
+    state,
+  }: {
+    interval?: number;
+    currentChainId: Hex;
+    currentTicker: string;
+    currentAddress: string;
+    disabled?: boolean;
+    tokenPricesService: AbstractTokenPricesService;
+    messenger: TokenRatesControllerMessenger;
+    state?: Partial<TokenRatesControllerState>;
+  }) {
+    super({
+      name: controllerName,
+      messenger,
+      state: { ...getDefaultTokenRatesControllerState(), ...state },
+      metadata,
+    });
 
-    this.defaultState = {
-      contractExchangeRates: {},
-      contractExchangeRatesByChainId: {},
-    };
-    this.initialize();
     this.setIntervalLength(interval);
-    this.getNetworkClientById = getNetworkClientById;
     this.#tokenPricesService = tokenPricesService;
+    this.#disabled = disabled;
+    this.#interval = interval;
+    this.#chainId = currentChainId;
```

Comment:
Similarly to `#selectedAddress`, would it be easier to take this value from `NetworkController` instead of storing it in a class variable?

Reply:
This pattern seems okay to me. In fact it may be advantageous to use private properties to cache values like this to mitigate synchronization issues. Like, if we have to pull a bunch of values from a bunch of controllers at the start of a polling iteration it might be better to do that all in one go and then cache those values rather than fetch them at various points during that iteration.